python/lsst/ctrl/oods/msgIngester.py

```diff
@@ -150,14 +161,19 @@ async def dequeue_and_ingest_files(self):
             for m in message_list:
                 if m.error():
                     if m.error().code() == KafkaError.UNKNOWN_TOPIC_OR_PART:
+                        if self.csc:
+                            self.csc.has_faulted = True
                         raise Exception("The topic or partition does not exist")
                     else:
+                        if self.csc:
+                            self.csc.has_faulted = True
```

Comment:
It looks to me that you always want to set `self.csc.has_faulted` to `True` if an error occurs, so why can''t you do it *before* checking the error code. In other words, is it possible to rewrite this as:

```python
if m.error():
    if self.csc:
        self.csc.has_faulted = True
    if m.error().code() == KafkaError.UNKNOWN_TOPIC_OR_PART:
        raise Exception("The topic or partition does not exist")
    else:
        raise Exception(f"KafkaError = {m.error().code()}")
```

Reply:
Good catch.  I added this in stages, and wasn't thinking. Fixed.